src/setfit/modeling.py

```diff
@@ -132,30 +140,28 @@ def forward(
 
         return outputs
 
-    def predict_proba(self, x_test: torch.Tensor) -> torch.Tensor:
+    def predict_proba(self, x_test: Union[torch.Tensor, "ndarray"]) -> Union[torch.Tensor, "ndarray"]:
+        is_tensor = isinstance(x_test, torch.Tensor)  # Otherwise assume it's ndarray
+        if not is_tensor:
+            x_test = torch.Tensor(x_test).to(self.device)
         self.eval()
 
-        return self(x_test)
+        out = self(x_test)
+        if not is_tensor:
+            return out.detach().cpu().numpy()
+        return out
 
     def predict(self, x_test: Union[torch.Tensor, "ndarray"]) -> Union[torch.Tensor, "ndarray"]:
-        is_tensor = isinstance(x_test, torch.Tensor)
-        if not is_tensor:  # then assume it's ndarray
-            x_test = torch.Tensor(x_test).to(self.device)
-
         probs = self.predict_proba(x_test)
 
-        if self.out_features == 1:
-            out = torch.where(probs >= 0.5, 1, 0)
+        if self.out_features == 1 or self.multitarget:
+            out = np.where(probs >= 0.5, 1, 0)
         else:
-            out = torch.argmax(probs, dim=-1)
-
-        if not is_tensor:
-            return out.cpu().numpy()
-
+            out = np.argmax(probs, dim=-1)
```

Comment:
`np.where` and `np.argmax` works with both ndarrays and tensors and I double checked that they return a tensor if given a tensor, making them more versatile than the `torch` variants